worker/src/main/scala/com/lucidchart/piezo/ConnectionProvider.scala

```diff
@@ -34,7 +34,8 @@ class ConnectionProvider(props: Properties) {
   @volatile
   private var cachedIpWithExpiration: Option[CachedIpWithExpiration] = None
 
-  private val pool: Pool = new Pool(getIP)
+  @volatile
```

Comment:
This is unnecessary since we only access it from within a synchronized block.

Reply:
We read from this value outside the synchronized block on lines 113, 115, and 139, for example to get a connection:
```
pool.connectionProvider.get.getConnection
```